Bug effect on "spigot server 1.7 "    "V1.4.0.009"

Hello, i have a proble with the version 1.4.0.009
So I will explain to you.
I have a lobby on my server it's just a vanilla with spigot 1.7
And when I give some effect to a player he will be kicked from the server.

Screenshot from the screen I have when I'm be kick. http://puu.sh/qxJhF/7f13bafccb.png

Pastebin from my client console: http://pastebin.com/2YJBR0fD

This problem doesn't happen with the version 1.4.0.008

In hope you can fix this, thanks for reading.

`[19:25:25] [Netty Client IO #3/INFO] [LiteLoader]: Baking listener list for PostLoginListener with 0 listeners
[19:25:25] [Netty Client IO #3/INFO] [FML]: Aborting client handshake "VANILLA"
[19:25:25] [Netty Client IO #3/INFO] [FML]: [Netty Client IO #3] Client side vanilla connection established
[19:25:25] [Netty Client IO #3/INFO] [NEIAddons]: Connected to new server, resetting commands
[19:25:26] [Client thread/INFO] [LiteLoader]: Baking listener list for PreJoinGameListener with 0 listeners
[19:25:26] [Client thread/INFO] [Armourer's Workshop]: Adding world access to world net.minecraft.client.multiplayer.WorldClient@37f8af3a
[19:25:26] [Client thread/INFO] [EnderCore]: Transforming Class [net.minecraft.inventory.ContainerRepair], Method [func_82848_d]
[19:25:26] [Client thread/INFO] [EnderCore]: Transforming net.minecraft.inventory.ContainerRepair Finished.
[19:25:29] [Client thread/INFO] [LiteLoader]: Baking listener list for JoinGameListener with 1 listeners
[19:25:29] [Netty Client IO #3/ERROR] [FML]: NetworkDispatcher exception
io.netty.handler.codec.DecoderException: java.lang.IndexOutOfBoundsException: readerIndex(9) + length(1) exceeds writerIndex(9): UnpooledHeapByteBuf(ridx: 9, widx: 9, cap: 9)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:263) ~[ByteToMessageDecoder.class:?]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:131) ~[ByteToMessageDecoder.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:337) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:323) [DefaultChannelHandlerContext.class:?]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:173) [ByteToMessageDecoder.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:337) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:323) [DefaultChannelHandlerContext.class:?]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [MessageToMessageDecoder.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:337) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:323) [DefaultChannelHandlerContext.class:?]
	at io.netty.handler.timeout.ReadTimeoutHandler.channelRead(ReadTimeoutHandler.java:149) [ReadTimeoutHandler.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:337) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:323) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:785) [DefaultChannelPipeline.class:?]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:100) [AbstractNioByteChannel$NioByteUnsafe.class:?]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:480) [NioEventLoop.class:?]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:447) [NioEventLoop.class:?]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:341) [NioEventLoop.class:?]
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101) [SingleThreadEventExecutor$2.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_101]
Caused by: java.lang.IndexOutOfBoundsException: readerIndex(9) + length(1) exceeds writerIndex(9): UnpooledHeapByteBuf(ridx: 9, widx: 9, cap: 9)
	at io.netty.buffer.AbstractByteBuf.checkReadableBytes(AbstractByteBuf.java:1160) ~[AbstractByteBuf.class:?]
	at io.netty.buffer.AbstractByteBuf.readByte(AbstractByteBuf.java:562) ~[AbstractByteBuf.class:?]
	at net.minecraft.network.PacketBuffer.readByte(SourceFile:496) ~[et.class:?]
	at net.minecraft.network.play.server.S1DPacketEntityEffect.func_148837_a(SourceFile:37) ~[in.class:?]
	at net.minecraft.util.MessageDeserializer.decode(SourceFile:40) ~[ez.class:?]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:232) ~[ByteToMessageDecoder.class:?]
	... 19 more
[19:25:29] [Netty Client IO #3/INFO] [THAUMCRAFT]: Restoring client configs.
[19:25:29] [Netty Client IO #3/INFO] [EnderCore]: Reset configs to client values for com.enderio.core.common.config.ConfigHandler
[19:25:29] [Netty Client IO #3/INFO] [EnderCore]: Reset configs to client values for tterrag.wailaplugins.config.WPConfigHandler
[19:25:29] [Client thread/INFO] [LiteLoader]: Baking listener list for ChatListener with 0 listeners
[19:25:29] [Netty Client IO #3/ERROR] [FML]: NetworkDispatcher exception
io.netty.handler.codec.DecoderException: java.lang.IndexOutOfBoundsException: readerIndex(9) + length(1) exceeds writerIndex(9): UnpooledHeapByteBuf(ridx: 9, widx: 9, cap: 9)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:263) ~[ByteToMessageDecoder.class:?]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:131) ~[ByteToMessageDecoder.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:337) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:323) [DefaultChannelHandlerContext.class:?]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:173) [ByteToMessageDecoder.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:337) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:323) [DefaultChannelHandlerContext.class:?]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [MessageToMessageDecoder.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:337) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:323) [DefaultChannelHandlerContext.class:?]
	at io.netty.handler.timeout.ReadTimeoutHandler.channelRead(ReadTimeoutHandler.java:149) [ReadTimeoutHandler.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:337) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:323) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:785) [DefaultChannelPipeline.class:?]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:100) [AbstractNioByteChannel$NioByteUnsafe.class:?]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:480) [NioEventLoop.class:?]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:447) [NioEventLoop.class:?]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:341) [NioEventLoop.class:?]
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101) [SingleThreadEventExecutor$2.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_101]
Caused by: java.lang.IndexOutOfBoundsException: readerIndex(9) + length(1) exceeds writerIndex(9): UnpooledHeapByteBuf(ridx: 9, widx: 9, cap: 9)
	at io.netty.buffer.AbstractByteBuf.checkReadableBytes(AbstractByteBuf.java:1160) ~[AbstractByteBuf.class:?]
	at io.netty.buffer.AbstractByteBuf.readByte(AbstractByteBuf.java:562) ~[AbstractByteBuf.class:?]
	at net.minecraft.network.PacketBuffer.readByte(SourceFile:496) ~[et.class:?]
	at net.minecraft.network.play.server.S1DPacketEntityEffect.func_148837_a(SourceFile:37) ~[in.class:?]
	at net.minecraft.util.MessageDeserializer.decode(SourceFile:40) ~[ez.class:?]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:232) ~[ByteToMessageDecoder.class:?]
	... 19 more`

--- a/src/main/java/am2/preloader/BytecodeTransformers.java
+++ b/src/main/java/am2/preloader/BytecodeTransformers.java
@@ -11,6 +11,10 @@ import org.objectweb.asm.tree.*;
 import java.lang.reflect.Field;
 import java.util.Iterator;
 
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+
 public class BytecodeTransformers implements IClassTransformer{
 	
 	@Override
@@ -744,6 +748,22 @@ public class BytecodeTransformers implements IClassTransformer{
 	      ClassReader cr = new ClassReader(bytes);
 	      ClassNode cn = new ClassNode();
 	      cr.accept(cn, 0);
+
+		//debug
+		if (!is_obfuscated){
+			try{
+				ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_MAXS);
+				cn.accept(cw);
+				File outDir = new File("asm/am2");
+				outDir.mkdirs();
+				DataOutputStream dout = new DataOutputStream(new FileOutputStream(new File(outDir, "SD1PacketEntityEffect_pre.class")));
+				dout.write(cw.toByteArray());
+				dout.flush();
+				dout.close();
+			}catch (Exception e){
+				e.printStackTrace();
+			}
+		}
 	      
 	      // Minecraft r1.7.10:
 	      // S1DPacketEntityEffect.java --> in.class
@@ -828,6 +848,10 @@ public class BytecodeTransformers implements IClassTransformer{
 	      
 	      String method1_searchinstruction_function = "readByte";
 	      String method1_searchinstruction_desc = "()B";
+
+		  // method used to check if receiving a vanilla or AM2 packet
+		  String method1_checkinstruction_function = "readableBytes";
+		  String method1_checkinstruction_desc = "()I";
 	      
 	      // replacement class is the same
 	      String method1_replaceinstruction_function = "readInt";
@@ -848,6 +872,9 @@ public class BytecodeTransformers implements IClassTransformer{
 	      
 	      String method2_searchinstruction_function = "writeByte";
 	      String method2_searchinstruction_desc = "(I)Lio/netty/buffer/ByteBuf;";
+
+//		  String method2_checkinstruction_function = "writableBytes";
+//		  String method2_checkinstruction_desc = "()I";
 	      
 	      // replacement class is the same
 	      // so is the instruction, but we'll leave it as being explicitly defined just in case
@@ -940,11 +967,38 @@ public class BytecodeTransformers implements IClassTransformer{
 			      }
 			      
 			      if (target != null && target2 != null){
-				      MethodInsnNode new_readint = new MethodInsnNode(Opcodes.INVOKEVIRTUAL, method1_searchinstruction_owner.getVal(is_obfuscated), method1_replaceinstruction_function, method1_replaceinstruction_desc);
-				      FieldInsnNode new_potion_id_variable = new FieldInsnNode(Opcodes.PUTFIELD, classname, potionID_intvar_name, potionID_intvar_desc);
-				      
-				      mn.instructions.set(target, new_readint);
-				      mn.instructions.set(target2, new_potion_id_variable);
+			      	  InsnList new_if = new InsnList();
+					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 1));
+					  new_if.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, method1_searchinstruction_owner.getVal(is_obfuscated), method1_checkinstruction_function, method1_checkinstruction_desc, false));
+					  new_if.add(new IntInsnNode(Opcodes.BIPUSH, 7));
+					  LabelNode elseLabel = new LabelNode();
+					  new_if.add(new JumpInsnNode(Opcodes.IF_ICMPNE, elseLabel));
+					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 0));
+					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 1));
+					  new_if.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, method1_searchinstruction_owner.getVal(is_obfuscated), method1_replaceinstruction_function, method1_replaceinstruction_desc, false));
+					  new_if.add(new FieldInsnNode(Opcodes.PUTFIELD, classname, potionID_intvar_name, potionID_intvar_desc));
+					  LabelNode endIfLabel = new LabelNode();
+					  new_if.add(new JumpInsnNode(Opcodes.GOTO, endIfLabel));
+					  new_if.add(elseLabel);
+					  new_if.add(new FrameNode(Opcodes.F_SAME, 0, null, 0, null));
+					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 0));
+					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 1));
+					  new_if.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, method1_searchinstruction_owner.getVal(is_obfuscated), method1_searchinstruction_function, method1_searchinstruction_desc, false));
+					  new_if.add(new FieldInsnNode(Opcodes.PUTFIELD, classname, potionID_intvar_name, potionID_intvar_desc));
+					  new_if.add(endIfLabel);
+					  new_if.add(new FrameNode(Opcodes.F_SAME, 0, null, 0, null));
+
+					  mn.instructions.insertBefore(target.getPrevious().getPrevious(), new_if);
+					  mn.instructions.remove(target.getPrevious().getPrevious()); // ALOAD 0
+					  mn.instructions.remove(target.getPrevious()); // ALOAD 1
+					  mn.instructions.remove(target);
+					  mn.instructions.remove(target2);
+
+//				      MethodInsnNode new_readint = new MethodInsnNode(Opcodes.INVOKEVIRTUAL, method1_searchinstruction_owner.getVal(is_obfuscated), method1_replaceinstruction_function, method1_replaceinstruction_desc);
+//				      FieldInsnNode new_potion_id_variable = new FieldInsnNode(Opcodes.PUTFIELD, classname, potionID_intvar_name, potionID_intvar_desc);
+//
+//				      mn.instructions.set(target, new_readint);
+//				      mn.instructions.set(target2, new_potion_id_variable);
 				      LogHelper.debug("Core: Success!  Replaced opcodes!");
 			      }
 		      } else if (mn.name.equals(method2_name.getVal(is_obfuscated)) && mn.desc.equals(method2_desc.getVal(is_obfuscated))){
@@ -968,9 +1022,37 @@ public class BytecodeTransformers implements IClassTransformer{
 			      }
 			      
 			      if (target != null && target2 != null){
+			      	  // I don't think this advanced packet logic is even needed, as a server running AM2 would require AM2 on the client anyway.
+//					  InsnList new_if = new InsnList();
+//					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 1));
+//					  new_if.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, method2_searchinstruction_owner.getVal(is_obfuscated), method2_checkinstruction_function, method2_checkinstruction_desc, false));
+//					  new_if.add(new IntInsnNode(Opcodes.BIPUSH, 7));
+//					  LabelNode elseLabel = new LabelNode();
+//					  new_if.add(new JumpInsnNode(Opcodes.IF_ICMPNE, elseLabel));
+//					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 1));
+//					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 0));
+//					  new_if.add(new FieldInsnNode(Opcodes.GETFIELD, classname, potionID_intvar_name, potionID_intvar_desc));
+//					  new_if.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, method2_searchinstruction_owner.getVal(is_obfuscated), method2_replaceinstruction_function, method2_replaceinstruction_desc, false));
+//					  LabelNode endIfLabel = new LabelNode();
+//					  new_if.add(new JumpInsnNode(Opcodes.GOTO, endIfLabel));
+//					  new_if.add(elseLabel);
+//					  new_if.add(new FrameNode(Opcodes.F_SAME, 0, null, 0, null));
+//					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 1));
+//					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 0));
+//					  new_if.add(new FieldInsnNode(Opcodes.GETFIELD, classname, potionID_intvar_name, potionID_intvar_desc));
+//					  new_if.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, method2_searchinstruction_owner.getVal(is_obfuscated), method2_searchinstruction_function, method2_searchinstruction_desc, false));
+//					  new_if.add(endIfLabel);
+//					  new_if.add(new FrameNode(Opcodes.F_SAME, 0, null, 0, null));
+//
+//					  mn.instructions.insertBefore(target.getPrevious().getPrevious(), new_if);
+//					  mn.instructions.remove(target.getPrevious().getPrevious());
+//					  mn.instructions.remove(target.getPrevious());
+//					  mn.instructions.remove(target);
+//					  mn.instructions.remove(target2);
+
 				      FieldInsnNode new_potion_id_variable = new FieldInsnNode(Opcodes.GETFIELD, classname, potionID_intvar_name, potionID_intvar_desc);
 				      MethodInsnNode new_writeint = new MethodInsnNode(Opcodes.INVOKEVIRTUAL, method2_searchinstruction_owner.getVal(is_obfuscated), method2_replaceinstruction_function, method2_replaceinstruction_desc);
-				      
+
 				      mn.instructions.set(target, new_potion_id_variable);
 				      mn.instructions.set(target2, new_writeint);
 				      LogHelper.debug("Core: Success!  Replaced opcodes!");
@@ -988,6 +1070,21 @@ public class BytecodeTransformers implements IClassTransformer{
 
 	      ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_MAXS);
 	      cn.accept(cw);
+
+		//debug
+		if (!is_obfuscated){
+			try{
+				File outDir = new File("asm/am2");
+				outDir.mkdirs();
+				DataOutputStream dout = new DataOutputStream(new FileOutputStream(new File(outDir, "SD1PacketEntityEffect_post.class")));
+				dout.write(cw.toByteArray());
+				dout.flush();
+				dout.close();
+			}catch (Exception e){
+				e.printStackTrace();
+			}
+		}
+
 	      return cw.toByteArray();
 	}
 	
@@ -1089,6 +1186,10 @@ public class BytecodeTransformers implements IClassTransformer{
 	      // change from readUnsignedByte to readInt
 	      String method1_searchinstruction_function = "readUnsignedByte";
 	      String method1_searchinstruction_desc = "()S";
+
+		  // method used to check if receiving a vanilla or AM2 packet
+		  String method1_checkinstruction_function = "readableBytes";
+		  String method1_checkinstruction_desc = "()I";
 	      
 	      // replacement class is the same
 	      String method1_replaceinstruction_function = "readInt";
@@ -1137,9 +1238,37 @@ public class BytecodeTransformers implements IClassTransformer{
 			      }
 			      
 			      if (target != null){
-				      MethodInsnNode new_readint = new MethodInsnNode(Opcodes.INVOKEVIRTUAL, method1_searchinstruction_owner.getVal(is_obfuscated), method1_replaceinstruction_function, method1_replaceinstruction_desc);
-				      
-				      mn.instructions.set(target, new_readint);
+			      	  FieldInsnNode fieldNode = (FieldInsnNode)target.getNext();
+					  InsnList new_if = new InsnList();
+					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 1));
+					  new_if.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, method1_searchinstruction_owner.getVal(is_obfuscated), method1_checkinstruction_function, method1_checkinstruction_desc, false));
+					  new_if.add(new IntInsnNode(Opcodes.BIPUSH, 8));
+					  LabelNode elseLabel = new LabelNode();
+					  new_if.add(new JumpInsnNode(Opcodes.IF_ICMPNE, elseLabel));
+					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 0));
+					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 1));
+					  new_if.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, method1_searchinstruction_owner.getVal(is_obfuscated), method1_replaceinstruction_function, method1_replaceinstruction_desc, false));
+					  new_if.add(fieldNode.clone(null));
+					  LabelNode endIfLabel = new LabelNode();
+					  new_if.add(new JumpInsnNode(Opcodes.GOTO, endIfLabel));
+					  new_if.add(elseLabel);
+					  new_if.add(new FrameNode(Opcodes.F_SAME, 0, null, 0, null));
+					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 0));
+					  new_if.add(new VarInsnNode(Opcodes.ALOAD, 1));
+					  new_if.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, method1_searchinstruction_owner.getVal(is_obfuscated), method1_searchinstruction_function, method1_searchinstruction_desc, false));
+					  new_if.add(fieldNode.clone(null));
+					  new_if.add(endIfLabel);
+					  new_if.add(new FrameNode(Opcodes.F_SAME, 0, null, 0, null));
+
+					  mn.instructions.insertBefore(target.getPrevious().getPrevious(), new_if);
+					  mn.instructions.remove(target.getPrevious().getPrevious()); // ALOAD 0
+					  mn.instructions.remove(target.getPrevious()); // ALOAD 1
+					  mn.instructions.remove(target);
+					  mn.instructions.remove(fieldNode); // PUTFIELD
+
+//				      MethodInsnNode new_readint = new MethodInsnNode(Opcodes.INVOKEVIRTUAL, method1_searchinstruction_owner.getVal(is_obfuscated), method1_replaceinstruction_function, method1_replaceinstruction_desc);
+//
+//				      mn.instructions.set(target, new_readint);
 				      LogHelper.debug("Core: Success!  Replaced opcodes!");
 			      }
 		      } else if (mn.name.equals(method2_name.getVal(is_obfuscated)) && mn.desc.equals(method2_desc.getVal(is_obfuscated))){

